src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -467,19 +471,42 @@ public SemanticVersion(string version)
         public SemanticVersion(int major, int minor, int patch, string preReleaseLabel, string buildLabel)
             : this(major, minor, patch)
         {
-            if (!string.IsNullOrEmpty(preReleaseLabel))
+            if (string.IsNullOrEmpty(preReleaseLabel) && string.IsNullOrEmpty(buildLabel))
             {
-                if (!Regex.IsMatch(preReleaseLabel, LabelUnitRegEx)) throw new FormatException(nameof(preReleaseLabel));
-
-                PreReleaseLabel = preReleaseLabel;
+                return;
```

Comment:
Since we already check three first parameters in base constructor I think it is better follow this and check preReleaseLabel and buildLabel separately too. We have no need to complicate.

If you see a need to be very strong here we should use the same approach in all constructors - build a string (make sense to use ToString() method) and check the string with IsValid() method.

Reply:
Makes sense to me, I'll make those changes and push when ready.